controllers/volumereplicationgroup_controller.go

```diff
@@ -312,10 +478,48 @@ func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumes(
 	return nil
 }
 
-// createVolumeReplicationCRForPVC creates a VolumeReplication CR with a PVC as its data source
-func (v *VolumeReplicationGroupReconciler) createVolumeReplicationCRForPVC(
+func (v *VolumeReplicationGroupReconciler) createVolumeReplicationResources(
+	ctx context.Context,
+	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
+	pvcList *corev1.PersistentVolumeClaimList) bool {
+	requeue := false
+
+	for _, pvc := range pvcList.Items {
+		// if the PVC is not bound yet, dont proceed.
+		if pvc.Status.Phase != corev1.ClaimBound {
+			log.Errorf("PVC is not yet bound status: %v", pvc.Status.Phase)
+
+			requeue = true
+
+			// continue processing other PVCs and return the need to requeue
+			continue
+		}
+
+		volRep := &volrep.VolumeReplication{}
+
+		err := v.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
+		if err != nil {
```

Comment:
Can reduce one level of nesting if we process the no-errors first. if handling the other changes can consider this as well.
```suggestion
		if err != nil {
```
```suggestion
		if err == nil {
		    continue
		    }
```